ZenPacks/zenoss/Microsoft/Windows/utils.py

```diff
@@ -362,11 +362,18 @@ def dumper(self, *args, **kwargs):
 Common datasource utilities.
 '''
 
-def checkExpiredPassword(config, events, error):
-    '''add password expired event'''
-    if 'Password expired' in error or 'Check username and password' in error:
+def errorMsgCheck(config, events, error):
+    """Check error message and generate appropriate event."""
+    if 'Password expired' in error:
         events.append({
-            'eventClass': '/Status/Winrm/Ping',
+            'eventClass': '/Status/Winrm/Auth/PasswordExpired',
```

Comment:
It's preferable to set an event's eventClassKey instead of eventClass when creating events. This allows the user far better control over the events if they're not satisfied with the default event class, severity, etc.

I would recommend setting eventClassKey appropriately here, and adding default event class mappings to bring them into the desired event classes by default.